src/vanilla/utils/splitAtom.ts

```diff
@@ -204,6 +208,13 @@ export function splitAtom<Item, Key>(
                   }
                   break
                 }
+                case 'bulk': {
+                  set(
+                    arrAtom,
+                    action.atoms.map((atom) => get(atom)),
```

Comment:
Hmm, this works if used properly. But, I think it's safer to find the index and create a new array based on it. Let's not assume the passed atoms are correct.

Reply:
In case of `reorder`, we can calculate `symmetrical difference` first and if it has zero length update the splitAtom.

```
case 'reorder': {
  const oldArr = get(arrAtom);
  const newArr = action.atoms;
  
  const difference = oldArr
    .filter(atom => !newArr.includes(atom))
    .concat(newArr.filter(atom => !oldArr.includes(atom)));

  if(difference.length === 0) {
    set(
      arrAtom,
      newArr.map((atom) => get(atom)),
    );
  }
  break
}
```

Reply:
My request is slightly different. I want to avoid using the atom value for the final array, only to find the index.

So, I would like to avoid this usage:
```js
const arrAtom = atom([1, 2, 3])
const atoms = splitAtom(arrAtom)

  set(atoms, { type: 'bulk', atoms: [atom(-999)] })
```
(This has too much power and we should probably update the original arr.)

Reply:
> only to find the index

Correct me if i am wrong.
In case of `bulk?`, using only indexes would means we should calculate moved, inserted, removed atoms separately. I do not understand what is good about it.

```
const newArrMapping = [
  // moved
  {
    oldIndex: n,
    newIndex: m
  },
  // inserted
  {
    oldIndex: null,
    newIndex: m
  },
  // removed
  {
    oldIndex: n,
    newIndex: null
  },
]
```

> const arrAtom = atom([1, 2, 3])
> const atoms = splitAtom(arrAtom)
> 
>   set(atoms, { type: 'bulk', atoms: [atom(-999)] })

I do not understand this either. What if 1, 2, 3 are removed and -999 is inserted? Because it is `bulk?` not `reorder` after all.

Reply:
okay then, I think I misunderstood `bulk`. I think what I want to support is `reorder` only.

It doesn't make much sense for me to insert arbitrarily items.

For "bulk", I'd update the original atom. We could add it as a shortcut, but my hesitation is it doesn't add any capability. We don't usually do it in Jotai codebase. 

If you need "bulk", please share your use case. It can be a doc issue.

Reply:
Thank you for your time.
Last question.

> I want to avoid using the atom value for the final array

I do not see any difference between `symmetrical difference with length zero` and `creating index mapping and construct an array from it`. Either way, we are creating an new array with exactly same items by comparing them. Is there any?
Or did i misunderstand it?

For the reordering operation name, if `reorder` is not good, how about `shuffle`. As in shuffling cards.

Reply:
`shuffle` to me sounds like random sorting.

What I have in mind with the plain array is this:
```js
const reorder = (array, indexList) =>
  indexList.map((index) => {
    if (index < 0 || index >= arr.length) {
      throw new Error('index out of bounds');
    }
    return array[index];
  });

// usage
const arr = ['a', 'b', 'c'];
const arr2 = reorder(arr, [2, 0])
```
The operation name `reorder` is tentative.